youtube_dl/extractor/facebook.py

```diff
@@ -451,14 +454,29 @@ def _lowercase_escape(s):
                    self._search_regex(r'ownerName"\s*:\s*"([^"]+)"', webpage, 'uploader', default=None) or \
                    self._og_search_title(webpage, default=None)
 
+        timestamp = self._search_regex(
+                r'datePublished":"(.+?)"', webpage,
+                'timestamp', default=None) or self._search_regex(
+                r'datePublished":"(.+?)"', tahoe_data.secondary,
+                'timestamp', default=None) or self._search_regex(
+                r'datePublished":"(.+?)"', tahoe_data.primary,
+                'timestamp', default=None)
+        timestamp = parse_iso8601(timestamp)
+
+        if timestamp == None and webpage.find('Paid Partnership') == -1 or\
+                (timestamp == None and webpage.find('Paid Partnership') > -1 and
+                 'cookiefile' in self._downloader.params):
+            timestamp = int_or_none(
```

Comment:
let's do it in two phases: 1. search_regex with or-s 2. take the result and use int_or_none